File: src/multi_world/python/example/third.py
from mpi4py import MPI
from abstraction import ServerManager, ClientManager

rank = MPI.COMM_WORLD.Get_rank()

if rank == 0:
  ServerManager.init()
  ServerManager.open()

  first_comm = ServerManager.accept()
  transfer_object = {'name': '0|Third  to 1|First', 'value': 3011}
  first_comm.send(transfer_object, 0)
  
  print('(Send) 0|Third  -> 1|First: ', end=' ')
  print(transfer_object)

  ServerManager.finalize()
elif rank == 1:
  ClientManager.init()

  second_comm = ClientManager.connect('second0')
  received_object = second_comm.recv(source=0)
  
  print('(Recv) 1|Third  <- 0|Second: ', end='')
  print(received_object)

  ClientManager.finalize()

File: src/multi_world/c/example/third.c
#include "libClientManager.h"
#include "libServerManager.h"
#include "message.h"

int main(int argc, char** argv) {
  MPI_Init(&argc, &argv);

  int rank;
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  Message message;
  MPI_Datatype MessageType;
  getMessageType(&MessageType);

  if(rank == 0) {
    const ServerManager* manager = Server_Init(argv, NULL, 1U);
    Server_Open(manager);
    const MPI_Comm* firstComm = Server_Accept(manager, NULL);

    fillMessage(&message, "0|Third  to 1|First", 3011);
    MPI_Send(&message, 1, MessageType, 0, 0, *firstComm);
    printMessage("(Send) 0|Third  -> 1|First: ", &message);

    Server_Finalize(manager);
  } else if(rank == 1) {
    const ClientManager* manager = Client_Init(1U);
    const MPI_Comm* secondComm = Client_Connect(manager, "second0", NULL);

    MPI_Recv(&message, 1, MessageType, 0, 0, *secondComm, MPI_STATUS_IGNORE);
    printMessage("(Recv) 1|Third  <- 0|Second:", &message);

    Client_Finalize(manager);
  }

  MPI_Finalize();
}